proto/v2/observation.proto

```diff
@@ -0,0 +1,68 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+syntax = "proto3";
+option go_package = "./proto";
+package datacommons.v2;
+
+import "stat.proto";
+
+message VariableObservation {
+  // Keyed by entity DCID
+  map<string, EntityObservation> by_entity = 1;
+}
+
+message EntityObservation {
+  // Facet are orderred by preference.
+  repeated FacetObservation by_facet = 1;
+}
+
+message FacetObservation {
+  string facet_id = 1;
+  // Observations are sorted by date
+  repeated datacommons.PointStat observations = 2;
+}
+
+// Generic observation request
+message V2ObservationRequest {
+  // A list of entity DCIDs
+  repeated string entities = 1;
+  // An arrow notation expression of entities
+  // Ex: country/USA<-containedInPlace{typeOf: State}
+  string entities_expr = 2;
```

Comment:
Are these a oneof?  Otherwise describe when entities vs. entities_expr should be set?

Reply:
It should typically be used as one of, but even specifying both are legal and the API can process. I am inclined to leave the freedom to set one or both.

Reply:
I see.  Should we expand _expr to _expression?  Every other field is expanded :)

Reply:
Done